gcloud/datastore/client.py

```diff
@@ -180,14 +181,15 @@ class Client(_BaseClient):
 
     def __init__(self, project=None, namespace=None,
                  credentials=None, http=None):
-        project = _determine_default_project(project)
-        if project is None:
-            raise EnvironmentError('Project could not be inferred.')
-        self.project = project
+        _ClientProjectMixin.__init__(self, project=project)
         self.namespace = namespace
         self._batch_stack = _LocalStack()
         super(Client, self).__init__(credentials, http)
 
+    @staticmethod
+    def _determine_default(project):
+        return _determine_default_project(project)
```

Comment:
@tseaver Care to take a stab this week? I don't have much time for non-reviews. (We could also get this in now and file an issue to improve code quality.)